### PR TITLE
Finalize Search Interface

### PR DESCRIPTION
**Pull request by jonseitz**
This PR rolls up the rest of the Search Interface, including the header above the results, the pagination, and the display of the number of results with the query searched. There's also a fair amount of cleanup and refactoring bundled in with this, as I found bugs that needed to be addressed on the way to getting the search interface in a testable state. 

The main files to focus on are `SearchHeader`, `SearchCount`, and `SearchPagination`, which are then added to the `App` component to compose the search interface inside of the `SearchWrapper`. I've also added a few end-to-end test that connect to the actual `test_SEC_DIRECTORY` index on Algolia to make sure that the full user flows work as expected.

To summarize most of the other changes that have been included here:
- Minor linting fixes (e.g. removing trailing whitespaces)
- Stylistic tweaks to get the full interface to fit on the screen
- Changing some click events to pointer events to prevent double-actions

Fixes: #19
Fixes: #6
Fixes: #5 
Fixes: #7

### CODE REVIEW

File: src/SearchCount/index.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  * Displays the total number of search hits, along with the query that was searched
+  */
+
+export const SearchCount = ({ searchResults }: Partial<StateResultsProvided>) => {
+  if (searchResults) {
+    const { nbHits, query } = searchResults;
+    return (
+     <>
+        <p className="search-pagination--count">
+          {`${nbHits} result${nbHits === 1 ? '' : 's'} for:`}
```
Comment (natalynyu) [inline src/SearchCount/index.tsx:13]: Out of curiosity, how is it printing "0 result**s** for:" when this is checking if `nbHits === 1`? I searched for "288" to see 0 results:
image

**Review by natalynyu — CHANGES_REQUESTED** [review]
Looks great! I just had a few small notes.

File: src/Keyboard/index.tsx [hunk 2]
```diff
@@ -73,18 +73,13 @@ const OnScreenKeyboard = ({
   const [keyboardCase, setKeyboardCase] = useState<CASE>(CASE.DEFAULT);
 
   /**
-   * Handles the logic for controlling the non-input keys, i.e shift, caps, and
-   * enter.
-   * The capitalization is not built into the keyboard library unfortunately,
-   * and I think that leaving out the upper/lower case functionality might be
-   * more confusing for users.
+   * Handles the logic for controlling the shift and caps-log keys. The
```
[the remark this answers is not in the document]
Comment (jonseitz) [reply src/Keyboard/index.tsx]: yep, that's what I meant. Updated in 317a44b

File: src/SearchCount/__tests__/SearchCount.test.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      render(<SearchCount searchResults={testSearchResults} />);
+      expect(screen.getByText('42 results for:')).toBeInTheDocument();
+    });
+  });
+  describe('With a blank query', function () {
+    const testSearchResults = {
+      nbHits: 42,
+      query: '',
+    } as SearchResults;
+    it('should show ""', function  () {
```
[the remark this answers is not in the document]
Comment (jonseitz) [reply src/SearchCount/__tests__/SearchCount.test.tsx]: I was splitting the tests for the "_n_ results for:" section away from the section that shows the query that was searched, since they're separate `<p>` elements, so this particular test isn't really concerned with how many results. I did add a little more clarification that it's specifically looking for "" in the query area in c6ac672

File: src/SearchCount/index.tsx (hunk 1 above)
Comment (jonseitz) [reply src/SearchCount/index.tsx:13]: The logic there is basically saying that if the `nbHits` value is anything **other than** `1`, it should show the `s` -- it doesn't really matter if the the value of `nbHits is less than or greater than `1`. Since we won't have negative or non-numeric values that's really the only situation where we'd need the word "result"; everywhere else we'd use "results"

Comment (natalynyu) [reply src/SearchCount/index.tsx:13]: Ah oops sorry I'm not sure what I was thinking! Thanks for clarifying.

**Review by natalynyu — APPROVED** [review]
Looks great! Approved.
